waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/metrics/MonitoredAspect.java

```diff
@@ -67,6 +71,12 @@ public Object monitor(ProceedingJoinPoint pjp, Monitored monitored) throws Throw
       increment(buildMetricPath(COUNTER, metricBasePath, getMonitorMetastore(), result));
```

Comment:
If we are always going to send both metrics (with and without tags), I don't see much reason for having two kind of methods, increment, incrementWithTags, submit, submitWithTags. 

We can just add the functionality to increment and submit methods. You can add the comment "Sends metrics with Tags: federation_namespace and method_name" in there.